File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.swift
var set = Set<String>()

let N = Int(readLine()!)!

for _ in 0..<N {
    let input = readLine()!.split(separator: " ").map { String($0) }
    if input[1] == "enter" { // enter면 insert
        set.insert(input[0]) //👉 O(1)
    } else { // leave면 remove
        set.remove(input[0]) //👉 O(1)
    }
}

set.sorted(by: >).forEach { name in
    print(name)
}

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.py
import sys
input = sys.stdin.readline

dct = {}

n = int(input().rstrip())

for _ in range(n):
    name, el = input().rstrip().split()
    if el == 'enter':
        dct[name] = True
    else:
        del dct[name]

print('\n'.join(sorted(dct.keys(), reverse=True)))